File: cpp/examples/error_handling.cpp
#include "twofinance/sdk_client.hpp"

#include <iostream>

int main() {
  twofinance::SdkConfig config;
  config.analytics_url = "https://analytics.example";

  twofinance::Transport transport = [](const twofinance::HttpRequest&) {
    return twofinance::HttpResponse{429, "{\"error\":\"rate_limited\"}"};
  };

  twofinance::SdkClient client(config, transport);
  try {
    client.analytics.indicators();
  } catch (const twofinance::ServiceError& error) {
    std::cout << "request failed with status " << error.status_code() << ": " << error.body() << "\n";
  }

  return 0;
}

File: php/examples/error_handling.php
<?php

declare(strict_types=1);

require __DIR__ . '/../src/Auth.php';
require __DIR__ . '/../src/TokenSource.php';
require __DIR__ . '/../src/StaticTokenSource.php';
require __DIR__ . '/../src/ServiceClient.php';
require __DIR__ . '/../src/SdkConfig.php';
require __DIR__ . '/../src/DomainClients.php';
require __DIR__ . '/../src/SdkClient.php';

use TwoFinance\Sdk\SdkClient;
use TwoFinance\Sdk\ServiceException;

$client = SdkClient::fromEnv();

try {
    $client->analytics->indicators();
} catch (ServiceException $exception) {
    echo sprintf(
        'request failed with status %d: %s%s',
        $exception->statusCode,
        $exception->body,
        PHP_EOL,
    );
}
